### ISSUE TITLE
Allow a destroy to proceed even if there are failures

### ISSUE THREAD

**Issue by lukehoban** [pulumi/pulumi#3304]
There are many cases where a user wants to do as much cleanup as possible, even if some resources can't succesfully be deleted.  We should offer a `--force` or similar flag on `pulumi destroy` to proceed with destroy operations even in the face of errors.  This may not be able to destroy dependent resources, but could destroy all unrelated resources.  It's possible this should even try to delete resources that are depended-on by undeletable resources - just in case it is possible (for example, if a Kubernetes resource fails to delete, we could try to proceed and delete the whole Kuberentes cluster).

Today - it is frequently the case when stacks are partly broken that they cannot be cleanly deleted, even with `pulumi refresh && pulumi destroy`.  This would provide a better effort option to do the best Pulumi can do to clean up resources under management of the stack.

```[tasklist]
### Tasks
- [ ] https://github.com/pulumi/pulumi/issues/15695
```

**Comment by lukehoban** [comment]
The features discussed in #3304 and #7747 would each solve this scenario as well, but via slightly more general mechanism.

**Comment by pgavlin** [comment]
@lukehoban #3304 is this issue... did you mean to link to something else?

**Comment by lukehoban** [comment]
> @lukehoban #3304 is this issue... did you mean to link to something else?

I believe this comment was intended for another issue which was a scenario that this and #7747 would address, but unfortunately I cannot now find what issue I was trying to reply on.

**Comment by mkosynskyi** [comment]
gcp:appengine:FlexibleAppVersion (drift-check):
    error: deleting urn:pulumi:dev::gcp-composer-airflow::gcp:appengine/flexibleAppVersion:FlexibleAppVersion::drift-check: 1 error occurred:
        * Error when reading or editing AppVersion: googleapi: Error 400: Cannot delete the final version of a service (module). Please delete the whole service (module) in order to delete this version.

**Comment by mkosynskyi** [comment]
When you want to delete something deployed on App Engine - you can't because App Engine wouldn't allow you to delete default service

**Comment by mkosynskyi** [comment]
const appService = new gcp.appengine.FlexibleAppVersion("drift-check", {
    project: projectId,
    versionId: "v1",
    entrypoint: {
        shell: "node /app/node.js",
    },
    runtime: "custom",
    envVariables: {
//        port: "8080",
//        MY_VARIABLE: "my-value",
        PULUMI_STACK_NAME: pulumiStack,
        PULUMI_ACCESS_TOKEN: pulumiAccessToken,
        SLACK_WEBHOOK_URL: slackwebhookurl,
        GOOGLE_APPLICATION_CREDENTIALS: googlecredentials,
    },
    deployment: {
        container: {
            image: image,
        },
    },
    automaticScaling: {
        coolDownPeriod: "120s",
        maxTotalInstances: 1,
        minTotalInstances: 0,
        cpuUtilization: {
            targetUtilization: 0.9,
        },
    },
//     manualScaling: {
//         instances: 1,
//     },
    resources: {
        cpu: 2,
        memoryGb: 4,
    },
    instanceClass: "F1",
    livenessCheck: {
        checkInterval: "30s",
        timeout: "4s",
        failureThreshold: 2,
        successThreshold: 1,
        initialDelay: "180s",
        path: "/health",
        host: "0.0.0.0",
    },
    readinessCheck: {
        checkInterval: "5s",
        timeout: "4s",
        failureThreshold: 2,
        successThreshold: 1,
        appStartTimeout: "300s",
        path: "/health",
        host: "0.0.0.0",
    },
    service: "drift-check",
    serviceAccount: customServiceAccount.email,
});

**Comment by justinvp** [comment]
Related: https://github.com/pulumi/pulumi/issues/14515

We're looking at adding `pulumi up --continue-on-error` and `pulumi destroy --continue-on-error`.

**Comment by tgummerer** [comment]
We implemented a `--continue-on-error` flag for `pulumi destroy` in https://github.com/pulumi/pulumi/pull/15727.  This was released in pulumi 3.112.0. That flag does everything that's planned here minus this sidenote:

>  It's possible this should even try to delete resources that are depended-on by undeletable resources

`--continue-on-error` will only delete resources that don't depend on the undeleteable resource.  This makes the operation safer to use, while still going ahead and destroying as much of the stack as possible without breaking any dependencies. 

We're tracking potentially introducing a `--force` flag in https://github.com/pulumi/pulumi/issues/15817.
